Remove Host discovery from core

kbrock opened the pull request:
- [x] https://github.com/ManageIQ/manageiq-ui-classic/pull/8626
- [x] https://github.com/ManageIQ/manageiq-providers-vmware/pull/856

jrafanie commented:
> * [ ]  `ManageIQ::Providers::Microsoft::Discovery`
> * [ ]  `ManageIQ::Providers::Openstack::Discovery`
> * [ ]  `ManageIQ::Providers::Ovirt::Discovery`
> * [ ]  `ManageIQ::Providers::Redhat::Discovery`
> * [ ]  `ManageIQ::Providers::Vmware::Discovery`

@kbrock we discussed and don't think we can remove those... let's start with IPMI and host discovery.

kbrock commented:
update:
- removed any impi deletion - this is a separate project
- removed network discovery. while this called into network discovery, that is still useful and will be used at a later time

Fryguy reviewed (approved):
LGTM - would like another set of eyes - @jrafanie or @agrare ?

Fryguy commented on app/models/host.rb:
```diff
@@ -705,185 +705,6 @@ def verify_credentials_with_ipmi(auth_type = nil)
-    return os_name, os_type
-  end
-
-  def detect_discovered_hypervisor(ost, ipaddr)
```
Note there's a subclass override of this method in the vmware repo: https://github.com/ManageIQ/manageiq-providers-vmware/blob/5e3dac8bd0b0ba09a5938ce8b6c5a2d3ace7e860/app/models/manageiq/providers/vmware/infra_manager/host.rb

agrare reviewed (approved):
:+1: LGTM

kbrock commented:
update:
- removed the function from vmware
